hxinsight-extension/src/integration-test/resources/rest/api/question-entity-resource-test-context.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!--
+  ~ #%L
+  ~ Alfresco HX Insight Connector
+  ~ %%
+  ~ Copyright (C) 2023 - 2024 Alfresco Software Limited
+  ~ %%
+  ~ This file is part of the Alfresco software.
+  ~ If the software was purchased under a paid Alfresco license, the terms of
+  ~ the paid license agreement will prevail.  Otherwise, the software is
+  ~ provided under the following open source license terms:
+  ~
+  ~ Alfresco is free software: you can redistribute it and/or modify
+  ~ it under the terms of the GNU Lesser General Public License as published by
+  ~ the Free Software Foundation, either version 3 of the License, or
+  ~ (at your option) any later version.
+  ~
+  ~ Alfresco is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  ~ GNU Lesser General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Lesser General Public License
+  ~ along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+  ~ #L%
+  -->
+<beans xmlns="http://www.springframework.org/schema/beans"
```

Comment:
WDYT about having this test config as Java annotation based config?
I think it would increase the readability and this case (tests) could be considered as not breaking the common repo/AMPs convention.
I am okay with XML config - that is just an idea we could discuss.

Reply:
I removed Integration test from this pr - I'll add it later